### PR TITLE
pre-release omnistaging cleanup

### PR DESCRIPTION
**Pull request by jakevdp**
Main changes:

- make `disable_omnistaging()` raise a warning
- add note in changelog
- modify docs to reflect this

### CODE REVIEW

**Comment by jakevdp** [comment]
I expect the "without omnistaging" CI should fail... once it does I'll sync past #6227

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ PLEASE REMEMBER TO CHANGE THE '..master' WITH AN ACTUAL TAG in GITHUB LINK.
     * `TraceContext` --> {func}`~jax.profiler.TraceAnnotation`
     * `StepTraceContext` --> {func}`~jax.profiler.StepTraceAnnotation`
     * `trace_function` --> {func}`~jax.profiler.annotate_function`
+  * Omnistaging can no longer be disabled. See [omnistaging](https://github.com/google/jax/blob/master/design_notes/prng.md)
```
Comment (mattjj) [inline CHANGELOG.md:24]: I think the file link might be to prng.md rather than omnistaging.md.

File: CHANGELOG.md [hunk 2]
```diff
@@ -272,7 +274,8 @@ PLEASE REMEMBER TO CHANGE THE '..master' WITH AN ACTUAL TAG in GITHUB LINK.
 
 * [GitHub commits](https://github.com/google/jax/compare/jax-v0.1.77...jax-v0.2.0).
 * Improvements:
-  * Omnistaging on by default. See {jax-issue}`#3370`.
+  * Omnistaging on by default. See {jax-issue}`#3370` and
```
Comment (mattjj) [inline CHANGELOG.md:277]: I think the file link might be to prng.md rather than omnistaging.md.

**Review by mattjj — APPROVED** [review]
LGTM! I think the PR comment may need to tweaked since we're not quite raising an error yet.

File: CHANGELOG.md (hunk 2 above)
Comment (jakevdp) [reply CHANGELOG.md:277]: Done, thanks!

**Pull request merged**